implementation/common/src/main/java/io/smallrye/jwt/algorithm/SignatureAlgorithm.java

```diff
@@ -31,6 +33,19 @@ public String getAlgorithm() {
     }
 
     public static SignatureAlgorithm fromAlgorithm(String algorithmName) {
-        return SignatureAlgorithm.valueOf(algorithmName);
+        try {
+            return SignatureAlgorithm.valueOf(algorithmName.toUpperCase());
+        } catch (Exception e) {
+            throw new IllegalArgumentException(
+                    "Invalid signature algorithm name: " + algorithmName + ", expected one of: " + getValidNames(), e);
+        }
+    }
+
+    private static String getValidNames() {
```

Comment:
We only need to make sure that if someone typed `rs256` or `RS256`, it works, so this is why to upper case conversion was there, and do a single specific `if` for `EdDSA`, this kind of check does not need a new dependency IMHO

Reply:
Jqwik is used here as well just not to mix it with `Junit.Parametrized`.

The true reason for jqwik is testing the method's property of failing for any other string than the valid ones [here](https://github.com/0rzech/smallrye-jwt/blob/e52d7f0bb542550a305e0c4fbacd80e4f9966e4e/implementation/common/src/test/java/io/smallrye/jwt/util/SignatureAlgorithmTest.java#L41-L44).

Reply:
Jqwik is used here as well just not to mix it with `Junit.Parametrized`.

The true reason for jqwik is testing the method's property of failing for any other string than the valid ones [here](https://github.com/0rzech/smallrye-jwt/blob/e52d7f0bb542550a305e0c4fbacd80e4f9966e4e/implementation/common/src/test/java/io/smallrye/jwt/util/SignatureAlgorithmTest.java#L41-L44) and most likely in other tests I have yet to write.